### PR TITLE
Introduce GitHub Actions

### PR DESCRIPTION
**Pull request by vezaynk**
Featuring:
 - CI with matrix of major react versions
 - ~~Automatically publish to NPM~~
 - ~~Enforce version bump on PRs (something by other PRs didn't follow)~~

### CODE REVIEW

**Review by IanVS — COMMENTED** [review]
Thanks for adding this.  I'd prefer not publishing to npm on merges, and keeping those manual, at least for now.  It gives a chance to batch up a few changes into a single release.  Or, if we want to do automatic releases, we should use something like https://github.com/intuit/auto, which allows controlling the type of release based on tags applied to the PR that is merged.

**Comment by vezaynk** [comment]
@IanVS We could leave it to run on `workflow-dispatch` only, so that it can easily be manually triggered. Or if there's no interest in this approach at all, I can remove all of them except the CI one.

**Comment by IanVS** [comment]
Let's just keep this PR to adding CI only, thanks.

**Comment by vezaynk** [comment]
@IanVS Done

**Comment by vezaynk** [comment]
Sample run: https://github.com/knyzorg/react-titled/actions/runs/3456682809

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - uses: actions/setup-node@v3
+    - run: yarn install --frozen-lockfile
+    - run: yarn build
+    - run: yarn build:examples
+
+  test:
+    strategy:
+      matrix:
+        react-version: 
+          - ^16.8.0
```
Comment (IanVS) [inline .github/workflows/ci.yml:35]: Let's leave off the caret on this, so that we actually test with 16.8.0, rather than the last 16.x version (16.14.0)

**Review by IanVS — COMMENTED** [review]
This is looking really good, thanks.  I have just one more small request, to be sure we actually test the minimum version.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vezaynk) [reply .github/workflows/ci.yml:35]: I would test both, actually.

Comment (IanVS) [reply .github/workflows/ci.yml:35]: I think there are not many changes between 16.14.0 and 17, but I'm not opposed to it.  We at least need 16.8.0, though.

**Comment by IanVS** [comment]
Thanks for this!  Would you mind taking a look at the failing action? https://github.com/tajo/react-titled/actions/runs/3463326286/jobs/5783397935#step:6:33

**Comment by vezaynk** [comment]
Yes. Ill take a look in a bit.

**Comment by vezaynk** [comment]
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

The new jsx transform is only supported 16.14+

There shouldn't be any breaking changes between the 16.8 and 16.14, but there may be bug fixes.

I don't think anyone should be targeting a minor version in their app. Maybe I can revert my last change and remove tests for the earliest version?

The alternative would be to use the old jsx transform, but its only a matter of time before it gets killed in a new major version.

Let me know what you think.

**Comment by IanVS** [comment]
Hm, I think I'm missing something.  Where are we using the new jsx transform?  All the files I looked at have an explicit `import React from 'react';` in them.

**Comment by vezaynk** [comment]
`import React from 'react';` isn't what decides which transform is used. It is part of the JSX transpiler (in this case managed by tsconfig I think).

Like I said, I can make it work with 16.8 but with no guarantee of not breaking future major releases. But as I said, I don't think anyone should be targeting 16.8.

If they are, they will bump it at the same time as installing/updating this package.

Up to you.
